src/mapper/cmd/main.go

```diff
@@ -237,16 +242,37 @@ func main() {
 
 	errgrp.Go(func() error {
 		defer errorreporter.AutoNotify()
+		go shutdownGracefullyOnCancel(errGroupCtx, metricsServer)
+
 		return metricsServer.Start(fmt.Sprintf(":%d", viper.GetInt(sharedconfig.PrometheusMetricsPortKey)))
 	})
 	errgrp.Go(func() error {
 		defer errorreporter.AutoNotify()
+		go shutdownGracefullyOnCancel(errGroupCtx, mapperServer)
+
 		return mapperServer.Start(":9090")
 	})
 
 	err = errgrp.Wait()
-	if err != nil && !errors.Is(err, http.ErrServerClosed) && !errors.Is(err, context.Canceled) {
-		logrus.WithError(err).Panic("Error when running server or HTTP server")
+	logrus.Infof("Network Mapper stopped")
+
+	if err != nil {
+		if !errors.Is(err, http.ErrServerClosed) && !errors.Is(err, context.Canceled) {
+			logrus.WithError(err).Error("failed to shutdown server")
+		}
+	}
+}
+
+func shutdownGracefullyOnCancel(errGroupCtx context.Context, server *echo.Echo) {
+	<-errGroupCtx.Done()
+	timeoutCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	shutdownErr := server.Shutdown(timeoutCtx)
+
+	if shutdownErr != nil {
+		logrus.WithError(shutdownErr).Error("failed to shutdown server")
 	}
 
+	_ = server.Close()
```

Comment:
oops, Close should be inside the `if`, and this should be a regular return with no error